tests/pipeline/test_buyback_auth.py

```diff
@@ -0,0 +1,511 @@
+"""
+Tests for the reference loader for EarningsCalendar.
```

Comment:
this test feels like it shares a lot with the earnings cal, do you think we can look at generalizing this for event sets?